src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -249,162 +236,25 @@ public async Task TestDisabledByDefaultAnalyzerEnabledWithEditorConfig(bool enab
             var exportProvider = workspace.Services.SolutionServices.ExportProvider;
 
             var service = Assert.IsType<DiagnosticAnalyzerService>(exportProvider.GetExportedValue<IDiagnosticAnalyzerService>());
-            var globalOptions = exportProvider.GetExportedValue<IGlobalOptionService>();
 
             var analyzer = service.CreateIncrementalAnalyzer(workspace);
 
             var syntax = false;
             var semantic = false;
 
-            // listen to events
-            service.DiagnosticsUpdated += (s, aCollection) =>
-            {
-                foreach (var a in aCollection)
-                {
-                    var diagnostics = a.Diagnostics;
-                    (syntax, semantic) = resultSetter(syntax, semantic, diagnostics);
-                }
-            };
-
             await analyzer.ForceAnalyzeProjectAsync(document.Project, CancellationToken.None);
 
-            // wait for all events to raised
-            await ((AsynchronousOperationListener)service.Listener).ExpeditedWaitAsync().ConfigureAwait(false);
-
-            // two should have been called.
-            Assert.Equal(expectedSyntax, syntax);
-            Assert.Equal(expectedSemantic, semantic);
-        }
-
-        [Fact]
-        public async Task TestOpenFileOnlyAnalyzerDiagnostics()
```

Comment:
this test isn't needed anymore.  LSP handles open/closed diagnostics through document/diagnostics and workspace/diagnostics.